pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -183,28 +191,28 @@ func niProbingUpdatePort(ctx *zedrouterContext, port types.NetworkPortStatus,
 // after port or NI changes, if we don't have a current uplink,
 // randomly assign one and publish, if we already do, leave it as is
 // each NI may have a different good uplink
-func checkNIprobeUplink(ctx *zedrouterContext, status *types.NetworkInstanceStatus) {
+func checkNIprobeUplink(ctx *zedrouterContext, status *types.NetworkInstanceStatus, intf string) {
 	// find and remove the stale info since the port has been removed
 	for _, info := range status.PInfo {
+		if info.IfName != intf {
+			continue
+		}
```

Comment:
If device network status initially comes with eth0, eth1 and eth2 as management, then PInfo will have eth0, eth1 and eth2.
Let's say eth2 is moved from mgmt to app-direct. Now device network status will only have eth0, eth1.
With the check above, will eth2 get deleted from PInfo?

Reply:
yes, it will. at this function run time, the PInfo has three map for eth0, eht1, and eth2. we reset the IsPresent at the beginning of the deviceNetworkInstance modify. and this 'eth2' will be the string pass into this function, and only the Pinfo of eth2 will pass this point and get deleted.

Reply:
But eth2 will not even be present in device network status. So we won't delete eth2 from PInfo.

Reply:
for the case of changing eth2 from Mgmt into App-Shared, the device Network status update  has the 'eth2'. We have tested this switching from mgmt->shared and back, right? Am I missing something?

Reply:
I'm asking about the case when eth2 is moved from mgmt to app-direct. Device nework status will not have eth2 in this case.